### PR TITLE
Fix tojson type

### PR DESCRIPTION
**Pull request by ganymedes01**

### CODE REVIEW

File: src/value-class.ts [hunk 1]
```diff
@@ -3,8 +3,7 @@ import { NamedTypeDefinition } from './reflection-type';
 export type Branded<A, BrandTag> = A & Brand<BrandTag>;
 
 export class Brand<B> {
-  // @ts-ignore
-  private valueClassBrand: B; // branding, DO NOT ACCESS
+  public readonly valueClassBrand!: B; // branding, DO NOT ACCESS
```
Comment (ganymedes01) [inline src/value-class.ts:6]: probably okay to make these public... 🤷‍♂️

Comment (sakari) [reply src/value-class.ts:6]: eh why? those are private for the express purpose of not allowing autocomplete

Comment (ganymedes01) [reply src/value-class.ts:6]: TypeScript loses the type unless the property is public. Same goes for the `valueClassShape`. And if a generic type isn't used anywhere in a class, it gets lost and can't be inferred anywhere where that class is used.

*.d.ts
```ts
// when public
export declare class Brand<B> {
    readonly valueClassBrand: B;
}
// when private
export declare class Brand<B> {
    readonly valueClassBrand;
}
```

Comment (ganymedes01) [reply src/value-class.ts]: Changed it to `protected` as it seems to keep the types in that case, but then it doesn't show up in autocomplete in most places 😬

Comment (sakari) [reply src/value-class.ts]: didn't we have this discussion a while ago?

Comment (ganymedes01) [reply src/value-class.ts]: I remember talking about this before, yeah 😅

**Pull request merged**